tests/test_electrolyzer.py

```diff
@@ -268,6 +319,133 @@ def test_electrolyzer_constant_efficiency(self):
+        # and 400 MW.
+        for idx in range(3):
+            np.testing.assert_allclose(
+                (results[f"Electrolyzer_fc66__line_{idx}_active"][idx]),
+                1.0,
+            )
+        # Check that the output massflow lies on the line segment
+        coef_a = solution.parameters(0)["Electrolyzer_fc66.a_eff_coefficient"]
+        coef_b = solution.parameters(0)["Electrolyzer_fc66.b_eff_coefficient"]
+        coef_c = solution.parameters(0)["Electrolyzer_fc66.c_eff_coefficient"]
+        a, b = solution._get_linear_coef_electrolyzer_mass_vs_epower_fit(
+            coef_a,
+            coef_b,
+            coef_c,
+            n_lines=3,
+            electrical_power_min=max(
+                solution.parameters(0)["Electrolyzer_fc66.minimum_load"],
+                0.01 * solution.bounds()["Electrolyzer_fc66.ElectricityIn.Power"][1],
+            ),
+            electrical_power_max=solution.bounds()["Electrolyzer_fc66.ElectricityIn.Power"][1],
+        )
+        for idx in range(3):
+            np.testing.assert_allclose(
+                results["Electrolyzer_fc66.Gas_mass_flow_out"][idx],
+                results["Electrolyzer_fc66.ElectricityIn.Power"][idx] * a[idx] + b[idx],
+            )
+        # Check hardcoded values
+        np.testing.assert_allclose(
+            results["Electrolyzer_fc66.Gas_mass_flow_out"],
+            [431.367058, 1285.95625642, 1673.61498453],
+            atol=1e-4,
+        )
+
+    def test_electrolyzer_equality_constraint_inactive_line(self):
+        """
+        This test is to check the functioning the example with an offshore wind farm in combination
+        with an electrolyzer and hydrogen storage. The electrolyzer is modelled as the option
+        LINEARIZED_THREE_LINES_EQUALITY.
+
+        Checks:
+        - Check that no line is active when the electrolyzer is switched off
+
+        """
+        import models.unit_cases_electricity.electrolyzer.src.example as example
+        from models.unit_cases_electricity.electrolyzer.src.example import (
+            MILPProblemEquality,
+        )
+
+        base_folder = Path(example.__file__).resolve().parent.parent
+
+        solution = run_esdl_mesido_optimization(
+            MILPProblemEquality,
+            base_folder=base_folder,
+            esdl_file_name="h2.esdl",
+            esdl_parser=ESDLFileParser,
+            profile_reader=ProfileReaderFromFile,
+            input_timeseries_file="timeseries_minimum_electrolyzer_power.csv",
+        )
+
+        results = solution.extract_results()
+
+        # Input power to the electrolyzer is below the minimum one,
+        # such that no line should be active
+        np.testing.assert_allclose(
+            (
+                results["Electrolyzer_fc66__line_0_active"][-1]
+                + results["Electrolyzer_fc66__line_1_active"][-1]
+                + results["Electrolyzer_fc66__line_2_active"][-1]
+            ),
+            0.0,
+        )
```

Comment:
I would also want to see the asset is switched on variable in this one, as well as a check that massflow and power are really still 0.

Reply:
Changed